src/core.js

```diff
@@ -1059,28 +1052,58 @@ define([
                             // when attempting a move, reset the state.  It will
                             // be changed again if the same move is attempted.
                             if (duplicateIsForMove) {
-                                _this.setUnsavedDuplicateNodeId(false);
+                                resetMethod(false);
                             }
                             _this.data.core.$modal.modal('hide');
-                            var input = _this.getCurrentMugInput("nodeID");
+                            var input = _this.getCurrentMugInput(inputID);
                             if (input) {
                                 input.select().focus();
                             }
                         }
                     },
                     {
-                        title: "Automatically rename to '" + newQuestionId + "'",
+                        title: "Automatically rename to '" + autoValue + "'",
                         cssClasses: 'btn-primary',
                         action: function () {
-                            mug.p.nodeID = newQuestionId;
-                            _this.setUnsavedDuplicateNodeId(false);
+                            mug.p[inputID] = autoValue;
+                            resetMethod(false);
                             _this.data.core.$modal.modal('hide');
                             _this.refreshVisibleData();
                             callback();
                         } 
                     }
-                
-                ]);
+                ];
+        }
+
+        if (this.data.core.hasXPathEditorChanged) {
+            this.alert(
+                "Unsaved Changes in Editor",
+                "You have UNSAVED changes in the Expression Editor. Please save "+
+                "changes before continuing.");
+            return false;
+        } else if (duplicate) {
+            var newQuestionId = this.data.core.form.generate_question_id(duplicate);
+
+            this.alert(
+                "Duplicate Question ID",
+                "'" + duplicate + "' " + verb + " the same Question ID as " +
+                "another question in the same group. Please change '" + 
+                duplicate + "' to a unique Question ID before continuing.",
+                alertDuplicateButtons(_this.setUnsavedDuplicateNodeId.bind(_this), "nodeID", newQuestionId)
+            );
+            return false;
+        } else if (duplicateChoice) {
```

Comment:
I'm not sure if it's legal to have a choice with a blank value, but I can't think of why it wouldn't be. In a quick test a choice didn't retain a blank value, but it also wasn't clear why, which may indicate a bug. Anyway, in the case that two choices have blank values this will evaluate to false even though it should evaluate to true, correct? Do you think that's worth fixing?